### PR TITLE
Add JinaEmbedding class 

### PR DESCRIPTION
**Pull request by JoanFM**
# Description

This PR adds JinaAIEmbedding class to allow users to obtain embeddings from Jina AI Embeddings API (https://jina.ai/embeddings/).

I am hesitating now how to add Documentation about it and I would like to hear suggestions from the team. I see there is a notebook using Jina Embeddings from HuggingFace but the usage is very different, should I edit that one or create a new one?

Fixes # (issue)

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Added new unit/integration tests
- [ ] Added new notebook (that tests end-to-end)
- [ ] I stared at the code and made sure it makes sense

# Suggested Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

**Comment by logan-markewich** [comment]
@JoanFM Thanks for the contribution! 

Yea I would add a new notebook using this embeddings class, and then link to the notebook in `docs/core_modules/model_modules/embeddings/modules.md`

File: docs/examples/embeddings/jinaai_embeddings.ipynb [hunk 1]
```diff
@@ -0,0 +1,383 @@
+{
```
Comment (ravi03071991) [inline docs/examples/embeddings/jinaai_embeddings.ipynb:1]: <p>The colab link seems pointing to cohereai notebook. Can you please correct it.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/run-llama/llama_index/pull/8704/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (ravi03071991) [inline docs/examples/embeddings/jinaai_embeddings.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #3.    # with input_typ='search_query'
</pre><p>I guess the comment is related to cohereai.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/run-llama/llama_index/pull/8704/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Comment (ravi03071991) [inline docs/examples/embeddings/jinaai_embeddings.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #1.    # with input_type = 'search_document'
</pre><p>Comment related to cohereai.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/run-llama/llama_index/pull/8704/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (ravi03071991) [inline docs/examples/embeddings/jinaai_embeddings.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #1.    llm = LiteLLM("command-nightly")
</pre><p>You also need to add 'COHERE_API_KEY' as environment variable to access their LLM via LiteLLM. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/run-llama/llama_index/pull/8704/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='22'/>

**Comment by JoanFM** [comment]
Hey @ravi03071991 ,

Thanks for the review, I put this as draft as I am still working on the notebook.

I will open it again as Ready for Review once it is finished. Thanks for the heads up though.

**Comment by JoanFM** [comment]
> Hey @ravi03071991 ,
> 
> Thanks for the review, I put this as draft as I am still working on the notebook.
> 
> I will open it again as Ready for Review once it is finished. Thanks for the heads up though.

Now it should be ready

**Comment by Disiok** [comment]
Thanks [JoanFM](https://github.com/JoanFM)! Could you resolve the merge conflict and the lint? Will put it to automerge after it's fixed!

**Review by Disiok — APPROVED** [review]

**Comment by JoanFM** [comment]
> Thanks [JoanFM](https://github.com/JoanFM)! Could you resolve the merge conflict and the lint? Will put it to automerge after it's fixed!

Done @Disiok , thanks

**Pull request merged**